static/js/reducers/email.js

```diff
@@ -15,38 +16,57 @@ import {
 } from '../actions';
 import type { Action } from '../flow/reduxTypes';
 import type {
-  Email,
-  EmailState
+  AllEmailsState,
+  EmailState,
+  EmailInputs,
 } from '../flow/emailTypes';
 
 export const INITIAL_EMAIL_STATE: EmailState = {
-  email:  {},
+  inputs: {},
   validationErrors: {},
   sendError: {},
 };
+export const INITIAL_ALL_EMAILS_STATE: AllEmailsState = {
+  searchResultEmail: INITIAL_EMAIL_STATE,
+  courseTeamEmail: INITIAL_EMAIL_STATE
+};
 
-export const NEW_EMAIL_EDIT: Email = {
+export const NEW_EMAIL_EDIT: EmailInputs = {
   subject:    null,
   body:       null,
-  query:      null,
 };
 
-export const email = (state: EmailState = INITIAL_EMAIL_STATE, action: Action) => {
+function updatedState(state: AllEmailsState, emailType: string, updateObject: Object) {
+  _.merge(state[emailType], updateObject);
+  return _.cloneDeep(state);
```

Comment:
actually, `return _.merge(_.cloneDeep(state), {[emailType]: updateObject});` would replace `state['searchResultEmal']` entirely instead of merging some limited set of properties into it. let me know what you think of the alternative i pushed